### ISSUE TITLE
Validation, including non-informative comparison with mmrm/SAS

### ISSUE THREAD

**Issue by kkmann** [openpharma/brms.mmrm#12]
It would probably be good to have a detailed comparison of a non-informative brms fit vs. at least mmrm to make sure brms is working fine here.

**Comment by wlandau** [comment]
* SBC
* Large well-behaved dataset and a small (maybe unbalanced) dataset
    * compare to `mmrm` (trust + verify that it already agrees with SAS)
    * render continuously in a vignette or unit test
    * one of those datasets should random dropouts and one-off missed visits
* other simulations to understand how it behaves (maybe separate from validation piece?)
    * compare decision-making behavior 
    * compare p-values to posterior probabilities
    * MAR vs not MAR, maybe MCAR?

**Comment by wlandau** [comment]
When we have a robust set of results, we might consider reaching out to CAMIS for their input on this project: https://psiaims.github.io/CAMIS/about.html

**Comment by wlandau** [comment]
Just finished a successful SBC study on the non-subgroup model, and the results are in a new vignette at https://openpharma.github.io/brms.mmrm/articles/sbc.html.

**Comment by wlandau** [comment]
Yesterday, @chstock and @yonicd and I discussed validation on example datasets. Maybe it would be good if `brms.mmrm` supported a special package function which accepts a dataset and returns a comparison of results between itself and the frequentist `mmrm` package. The output could be a compact table like the one in https://openpharma.github.io/brms.mmrm/articles/comparison.html#comparison. We could create a new vignette to run several comparisons like these, or maybe we could append these analyses to https://openpharma.github.io/brms.mmrm/articles/comparison.html itself.

**Comment by wlandau** [comment]
> Maybe it would be good if brms.mmrm supported a special package function which accepts a dataset and returns a comparison of results between itself and the frequentist mmrm package.

When I originally suggested this, I forgot how much ad hoc text manipulation was required to match up the parameters between models. After looking back at the comparison vignette, I no longer think it is feasible. But I can write another vignette for the BCVA dataset from `mmrm`.

**Comment by wlandau** [comment]
> other simulations to understand how it behaves (maybe separate from validation piece?)
> * compare decision-making behavior
> * compare p-values to posterior probabilities
> * MAR vs not MAR, maybe MCAR?

Looking back at this comment, I think that might still be interesting to explore outside the package, but the vignettes in the package are getting quite busy, and that kind of testing might have to do with understanding the model more than actual validation. I think we will be on good footing with simulation-based calibration and real data comparisons with `mmrm` on the datasets from https://openpharma.github.io/mmrm/latest-tag/articles/mmrm_review_methods.html.

**Comment by chstock** [comment]
> > Maybe it would be good if brms.mmrm supported a special package function which accepts a dataset and returns a comparison of results between itself and the frequentist mmrm package.
> 
> When I originally suggested this, I forgot how much ad hoc text manipulation was required to match up the parameters between models. After looking back at the comparison vignette, I no longer think it is feasible. But I can write another vignette for the BCVA dataset from `mmrm`. That way we have FEV1 and BCVA covered, which are the exact datasets that `mmrm` uses when comparing with SAS.

Sounds good. Happy to draft or review, whatever you prefer.

**Comment by wlandau** [comment]
Thanks @chstock! I will take the time to write the vignette (adapting your FEV1 vignette) and tag you for review.

**Comment by wlandau** [comment]
We now have 3 publicly available vignettes to support validation:

1. https://openpharma.github.io/brms.mmrm/articles/sbc.html: simulation-based calibration to ensure the model is implemented correctly (the posterior distribution estimated by `brms` agrees with the prior distribution simulated custom R code in https://github.com/openpharma/brms.mmrm/tree/main/sbc.
2. An example analysis comparing `mmrm` to `brms.mmrm` on the FEV1 dataset from the `mmrm` package: https://openpharma.github.io/brms.mmrm/articles/fev1.html. Parameter estimates agree well. 
3. Same as (2), but using the much larger BCVA dataset from the `mmrm` package: https://openpharma.github.io/brms.mmrm/articles/bcva.html. Likewise, there is strong agreement between `mmrm` and `brms.mmrm`.

(1) is state-of-the-art for Bayesian methods in general, and (2) and (3) together the same workflow and datasets that `mmrm` uses to compare itself with SAS: https://openpharma.github.io/mmrm/latest-tag/articles/mmrm_review_methods.html.

We talked about other areas to explore, such as power and decision-making in a clinical trial, but on reflection I feel these investigations are separate from validation and belong outside `brms.mmrm`.

In summary, I think we can close this issue. Do others agree?

**Comment by yonicd** [comment]
looks good! 

just to verify - both the fev and bcva are not real data from trials. they are both simulated data. where bcva is reproducible and fev isnt atm.

**Comment by wlandau** [comment]
Thanks for the clarification on the datasets, @yonicd. I think we can always run the same comparisons on real datasets when the opportunity presents itself.

Also, I started CAMIS discussion thread: https://github.com/PSIAIMS/CAMIS/discussions/151. They seem to be focused on frequentist methods and R vs SAS, and hopefully the discussion will help us find out if it makes sense to post to CAMIS.

**Comment by chstock** [comment]
Thanks, @wlandau. I agree, the issue can be closed. 

One last thing that I would perhaps find helpful from a user perspective is if there were a comment/ summary on the state of the validation somewhere in the documentation. This could for example be in the README file, which currently states that the package "_has not been validated for serious analyses_". This may even still be the case, but given the SBC and the two examples sounds a bit too modest to me now. And, could we make recommendations on what may be a good idea for further validation to use the package for serious analyses?

**Comment by wlandau** [comment]
From @yonicd, c.f. https://github.com/stan-dev/posteriordb

**Comment by wlandau** [comment]
> One last thing that I would perhaps find helpful from a user perspective is if there were a comment/ summary on the state of the validation somewhere in the documentation. This could for example be in the README file, which currently states that the package "has not been validated for serious analyses". This may even still be the case, but given the SBC and the two examples sounds a bit too modest to me now. And, could we make recommendations on what may be a good idea for further validation to use the package for serious analyses?

Great catch @chstock, implemented in 2564ddc5ad3dc6c3b2d80a10feb7aa8bb8a38c9e.
